tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -77,167 +56,66 @@ setup() {
-    set_metadata_annotation "$kata_pod_with_nydus_config" \
-        "io.containerd.cri.runtime-handler" \
-        "kata-${KATA_HYPERVISOR}"
+    pod_config="${pod_config_dir}/pod-guest-pull-in-trusted-storage.yaml"
+    storage_config="${pod_config_dir}/confidential/trusted-storage.yaml"
+    local_device=$(create_loop_device "/tmp/trusted-storage.img")
+    sed -i "s/RUNTIMECLASS/kata-${KATA_HYPERVISOR}/g" $pod_config
+    sed -i "s/NODE_NAME/$node/g" $pod_config
+    sed -i "s/NODE_NAME/$node/g" $storage_config
+    sed -i "s|LOCAL_DEVICE|$local_device|g" $storage_config
 
     # For debug sake
-    echo "Pod $kata_pod_with_nydus_config file:"
-    cat $kata_pod_with_nydus_config
-
-    add_allow_all_policy_to_yaml "$kata_pod_with_nydus_config"
-    k8s_create_pod "$kata_pod_with_nydus_config"
+    echo "Trusted storage $storage_config file:"
+    cat $storage_config
+    echo "Pod $pod_config file:"
+    cat $pod_config
 
-    echo "Kata pod test-e2e with nydus annotation is running"
-    echo "Checking the image was pulled in the guest"
-
-    sandbox_id=$(get_node_kata_sandbox_id $node)
-    echo "sandbox_id is: $sandbox_id"
-    # With annotation for nydus, only rootfs for pause container can be found on host
-    assert_rootfs_count "$node" "$sandbox_id" "1"
+    # Create persistent volume and persistent volume claim
+    kubectl create -f $storage_config
 
-    kubectl delete -f $kata_pod_with_nydus_config
-
-    # 2. Create one kata pod with the $unencrypted_image_1 image and without nydus annotation
-    kata_pod_without_nydus_config="$(new_pod_config "$unencrypted_image_1" "kata-${KATA_HYPERVISOR}")"
-    set_node "$kata_pod_without_nydus_config" "$node"
-    set_container_command "$kata_pod_without_nydus_config" "0" "sleep" "30"
-
-    # For debug sake
-    echo "Pod $kata_pod_without_nydus_config file:"
-    cat $kata_pod_without_nydus_config
-
-    add_allow_all_policy_to_yaml "$kata_pod_without_nydus_config"
-    k8s_create_pod "$kata_pod_without_nydus_config"
-
-    echo "Kata pod test-e2e without nydus annotation is running"
-    echo "Check the image was not pulled in the guest"
-    sandbox_id=$(get_node_kata_sandbox_id $node)
-    echo "sandbox_id is: $sandbox_id"
-
-    # The assert_rootfs_count will be FAIL.
-    # The expect count of rootfs in host is "2" but the found count of rootfs in host is "1"
-    # As the the first time we pull the $unencrypted_image_1 image via nydus-snapshotter in the guest
-    # for all subsequent pulls still use nydus-snapshotter in the guest
-    # More details: https://github.com/kata-containers/kata-containers/issues/8337
-    # The test case will be PASS after we use containerd 2.0 with 'image pull per runtime class' feature:
-    # https://github.com/containerd/containerd/issues/9377
-    assert_rootfs_count "$node" "$sandbox_id" "2"
-}
-
-@test "Test we can pull an other unencrypted image outside the guest and then inside the guest successfully" {
```

Comment:
Maybe we should keep this test?

Reply:
@ChengyuZhu6 , mind to explain why this and `"Test we can pull an unencrypted image inside the guest twice in a row and then outside the guest successfully" ` aren't needed anymore?